internal/slices.go

```diff
@@ -0,0 +1,22 @@
+package internal
```

Comment:
The functions here seem like good candidates for `go-collections`, if we like the ergonomics and can decide on appropriate names.

Reply:
Could we use https://pkg.go.dev/golang.org/x/exp/slices#DeleteFunc in place of this remove function?

re: map and filter, yeah we could add that to collections, but I think there are some existing candidates to consider first https://github.com/samber/lo

Reply:
My Google-fu failed pretty miserably on this one; I definitely don't want to be reinventing the wheel!